src/dom-wrapper.ts

```diff
@@ -10,19 +10,23 @@ export class DOMWrapper<ElementType extends Element> {
     this.element = element
   }
 
-  classes(className?: string) {
+  classes(): string[]
+  classes(className: string): boolean
+  classes(className?: string): string[] | boolean {
     const classes = this.element.classList
 
     if (className) return classes.contains(className)
 
     return Array.from(classes)
   }
 
-  attributes(key?: string) {
+  attributes(): { [key: string]: string }
+  attributes(key: string): string
+  attributes(key?: string): { [key: string]: string } | string {
     const attributes = this.element.attributes
-    const attributeMap = {}
+    const attributeMap: { [key: string]: string } = {}
```

Comment:
I tend to do `Record<string, string>`; is there any benefit to one or the other?

Reply:
`Record<string, string>` implementation is pretty much that:
```ts
type Record<K extends keyof any, T> = {
    [P in K]: T;
};
```

Using this syntax allows greater control on the value, in this case `Record<string, string>` is clearer in this case

Reply:
I agree, can we make this change @cexbrayat ? thanks!

Reply:
Sure, done!